src/unity/lib/unity_sframe.cpp

```diff
@@ -1950,6 +1950,186 @@ void unity_sframe::explore(const std::string& path_to_client, const std::string&
       }
     };
 
+    auto getAccordian = [self, &reader, &ew, &column_names, &empty_tz](std::string column_name, size_t index) {
+        ASSERT_TRUE(std::find(column_names.begin(), column_names.end(), column_name) != column_names.end());
+        DASSERT_LT(index, self->size());
+        DASSERT_GE(index, 0);
+
+        auto modal_sa = self->select_column(column_name);
+        auto gl_sa = gl_sarray(modal_sa);
+
+        flexible_type value = gl_sa[index];
+
+        switch (value.get_type()) {
+          case flex_type_enum::UNDEFINED:
+            break;
+          case flex_type_enum::FLOAT:
+            {
+              std::stringstream ss;
+              ss << "{\"modal_spec\": {\"index\": " << index << ", \"column\":" << turi::visualization::extra_label_escape(column_name);
```

Comment:
Similarly, I don't think we should call this `modal_spec` - the UI may evolve over time and may or may not be modal. I think all of the metadata here could be top-level within this dictionary, as:
```
{
    "index": ...,
    "column": ...,
    "type": ...,
    "data": ...
}
```